File: app/src/main/java/com/utvgo/handsome/diff/ITVBox.java
package com.utvgo.handsome.diff;

import android.content.Context;

public interface ITVBox {

    interface FetchUrlByVODAssetIdCallBack
    {
        void onReceivedUrl(final String vodId, final String url);
    }

    String getCA(final Context context);
    String getRegionCode(final Context context);
    void initDeviceInfo(final Context context);

    void fetchUrlByVODAssetId(final Context context, final String vodId, final FetchUrlByVODAssetIdCallBack callback);
}

File: app/src/main/java/com/utvgo/handsome/diff/ITVBox.kt
package com.utvgo.handsome.diff

import android.content.Context


interface ITVBox {
    interface FetchUrlByVODAssetIdCallBack {
        fun onReceivedUrl(vodId: String?, url: String?)
    }

    fun getCA(context: Context?): String?
    fun getRegionCode(context: Context?): String?
    fun getLoginToken(context: Context?): String?
    fun getDeviceModel(context: Context?): String?
    fun getCookies(context: Context?): String?
    fun initDeviceInfo(context: Context?)
    fun getUserId(context: Context?): String?
    fun exit(context: Context?)
    fun fetchUrlByVODAssetId(
        context: Context?,
        vodId: String?,
        callback: FetchUrlByVODAssetIdCallBack?
    )
}